src/api/providers/cerebras.ts

```diff
@@ -0,0 +1,329 @@
+ * Also strips thinking tokens from assistant messages to prevent model confusion.
+ */
+function convertToCerebrasMessages(openaiMessages: any[]): Array<{ role: string; content: string }> {
+	return openaiMessages
+		.map((msg) => {
+			let content = flattenMessageContent(msg.content)
+
+			// Strip thinking tokens from assistant messages to prevent confusion
+			if (msg.role === "assistant") {
+				content = stripThinkingTokens(content)
+			}
+
+			return {
+				role: msg.role,
+				content,
+			}
+		})
+		.filter((msg) => msg.content.trim() !== "") // Remove empty messages
+}
+
+export class CerebrasHandler extends BaseProvider implements SingleCompletionHandler {
+	private apiKey: string
+	private providerModels: typeof cerebrasModels
+	private defaultProviderModelId: CerebrasModelId
+	private options: ApiHandlerOptions
+	private lastUsage: { inputTokens: number; outputTokens: number } = { inputTokens: 0, outputTokens: 0 }
+
+	constructor(options: ApiHandlerOptions) {
+		super()
+		this.options = options
+		this.apiKey = options.cerebrasApiKey || ""
+		this.providerModels = cerebrasModels
+		this.defaultProviderModelId = cerebrasDefaultModelId
+
+		if (!this.apiKey) {
+			throw new Error("Cerebras API key is required")
+		}
+	}
+
+	getModel(): { id: CerebrasModelId; info: (typeof cerebrasModels)[CerebrasModelId] } {
+		const modelId = (this.options.apiModelId as CerebrasModelId) || this.defaultProviderModelId
+		return {
+			id: modelId,
+			info: this.providerModels[modelId],
+		}
+	}
+
+	async *createMessage(
+		systemPrompt: string,
+		messages: Anthropic.Messages.MessageParam[],
+		metadata?: ApiHandlerCreateMessageMetadata,
+	): ApiStream {
+		const {
+			id: model,
+			info: { maxTokens: max_tokens },
+		} = this.getModel()
+		const temperature = this.options.modelTemperature ?? CEREBRAS_DEFAULT_TEMPERATURE
+
+		// Convert Anthropic messages to OpenAI format, then flatten for Cerebras
+		// This will automatically strip thinking tokens from assistant messages
+		const openaiMessages = convertToOpenAiMessages(messages)
+		const cerebrasMessages = convertToCerebrasMessages(openaiMessages)
+
+		// Prepare request body following Cerebras API specification exactly
+		const requestBody = {
+			model,
+			messages: [{ role: "system", content: systemPrompt }, ...cerebrasMessages],
+			stream: true,
+			// Use max_completion_tokens (Cerebras-specific parameter)
+			...(max_tokens && max_tokens > 0 && max_tokens <= 32768 ? { max_completion_tokens: max_tokens } : {}),
+			// Clamp temperature to Cerebras range (0 to 1.5)
+			...(temperature !== undefined && temperature !== CEREBRAS_DEFAULT_TEMPERATURE
+				? {
+						temperature: Math.max(0, Math.min(1.5, temperature)),
+					}
+				: {}),
+		}
+
+		try {
+			const response = await fetch(`${CEREBRAS_BASE_URL}/chat/completions`, {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+					Authorization: `Bearer ${this.apiKey}`,
+					"User-Agent": "roo-cline/1.0.0",
```

Comment:
For what it's worth we have a DEFAULT_HEADERS that contains a "User-Agent" if you want to reuse that